setup.sh

```
@@ -231,18 +231,23 @@ function _main
 
   IMAGE_NAME=${INFO%;*}
   CONTAINER_NAME=${INFO#*;}
+  # echo "DEBUG: INFO=${INFO}"
+  # echo "DEBUG: PRE_IMAGE_NAME=${IMAGE_NAME}"
+  # echo "DEBUG: PRE_CONTAINER_NAME=${CONTAINER_NAME}"
 
   if [[ -z ${IMAGE_NAME} ]]
   then
     if [[ ${CRI} == "docker" ]]
     then
-      IMAGE_NAME=tvial/docker-mailserver:latest
+      IMAGE_NAME=${NAME}
```

Better to use `${NAME?}` here, to fail early and loud.

---

Worth a change to look at, again per comment above, NAME should always be present

---

```suggestion
      IMAGE_NAME=${NAME:-tvial/docker-mailserver:latest}
```

---

@williamdes I am not sure if the default value should be set at this depth. Come to think of it, I am not even sure if `NAME` should be used as an interface here, that will spell trouble somewhere down the line due to name clashes.

---

I did end up changing as @williamdes suggested, if it reaches that far into the execution without NAME set, it does either need a default value or to bomb. Again, open to suggestions.